src/common/snippets/src/lowered/pass/serialize_control_flow.cpp

```diff
@@ -9,16 +9,22 @@
 #include "snippets/lowered/linear_ir.hpp"
 #include "snippets/op/serialization_node.hpp"
 #include "snippets/snippets_isa.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/lowered/linear_ir_builder.hpp"
 
 namespace ov {
 namespace snippets {
 namespace lowered {
 namespace pass {
 
-bool SerializeControlFlow::run(LinearIR& linear_ir) {
+bool SerializeControlFlow::run(const LinearIR& original_linear_ir) {
     OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::SerializeControlFlow")
-    if (linear_ir.empty())
+    if (original_linear_ir.empty())
         return false;
+    const auto& linear_ir = m_update_dynamic_ops ? LinearIRBuilder().clone(original_linear_ir) : original_linear_ir;
+
+    const auto& loop_manager = linear_ir.get_loop_manager();
+    const auto& loop_info_map = loop_manager ? loop_manager->get_map() : std::map<size_t, LoopInfoPtr>{};
```

Comment:
Can `loop_manager` be empty? If it can be, could you please provide examples? I'm not sure it's safety and we should fix it.

Reply:
Got it, as I can see, it's more for internal and service functions. Thank you for the explanation